### PR TITLE
feat: add VR controls to demo

### PR DESCRIPTION
**Pull request by WoodNeck**
## Issue
#297 

## Details
Added "enter VR" button to previous demos.
I've also fixed the issue that animation not works while I'm working on it.

### CODE REVIEW

File: test/unit/utils/math-util.spec.js [hunk 1]
```diff
@@ -1,89 +1,160 @@
...
+				vec3.normalize(vec3.create(), vec3.fromValues(0, 0.5, 1)),
+			];
+			const expects = [
+				0,
+				-Math.PI / 2,
+				Math.PI / 2,
+				-Math.PI,
+			];
+
+			// When
```
Comment (daybrush) [inline test/unit/utils/math-util.spec.js]: @WoodNeck 

It's When, Then?

Comment (WoodNeck) [reply test/unit/utils/math-util.spec.js]: Looks like I've forgotten to add `Then`, I'll add it.

File: src/PanoViewer/PanoViewer.js [hunk 2]
```diff
@@ -170,6 +170,10 @@ export default class PanoViewer extends Component {
 			return this;
 		}
 
+		// Check XR support at not when imported, but when created.
+		// This is for easier use of polyfills
+		checkXRSupport();
```
Comment (daybrush) [inline src/PanoViewer/PanoViewer.js]: @WoodNeck 

Is there an error if called more than once?

If so, you need to retest. If not, there is no need to recheck.

**Comment by daybrush** [comment]
It may be necessary to check whether or not userAgent is present. Error in node environment

File: src/PanoViewer/PanoViewer.js (hunk 2 above)
Comment (WoodNeck) [reply src/PanoViewer/PanoViewer.js]: I don't think that there can be an error when it called multiple times.
But, you're right that there's no need to check it again.

**Pull request merged**